src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -399,10 +401,108 @@ void generateDeclarations() {
     }
   }
 
+  private boolean isGoogCall(Node node) {
+    if (!node.isCall()) {
+      return false;
+    }
+
+    if (node.getFirstChild() == null || !node.getFirstChild().isGetProp()) {
+      return false;
+    }
+
+    Node getPropNode = node.getFirstChild();
+
+    if (getPropNode.getFirstChild() == null || !getPropNode.getFirstChild().isName()) {
+      return false;
+    }
+
+    return getPropNode.getFirstChild().getString().equals("goog");
+  }
+
+  private boolean isGoogRequire(Node node) {
+    if (!isGoogCall(node)) {
+      return false;
+    }
+
+    if (node.getFirstChild() == null) {
+      return false;
+    }
+
+    Node getPropNode = node.getFirstChild();
+
+    if (getPropNode.getChildAtIndex(1) == null || !getPropNode.getChildAtIndex(1).isString()) {
+      return false;
+    }
+    return getPropNode.getChildAtIndex(1).getString().equals("require");
+  }
+
+  private boolean isGoogModule(Node node) {
+    if (!isGoogCall(node)) {
+      return false;
+    }
+    if (node.getFirstChild() == null) {
+      return false;
+    }
+    Node getPropNode = node.getFirstChild();
+    if (getPropNode.getChildAtIndex(1) == null || !getPropNode.getChildAtIndex(1).isString()) {
+      return false;
+    }
+    return getPropNode.getChildAtIndex(1).getString().equals("module");
+  }
+
+  private boolean isGoogRequireAssignment(Node node) {
+    if (!(node.isConst() || node.isVar() || node.isLet())) {
+      return false;
+    }
+
+    if (node.getFirstFirstChild() == null) {
+      return false;
+    }
+
+    return isGoogRequire(node.getFirstFirstChild());
+  }
+
+  private String getGoogModuleId(Node node) {
+    if (isGoogModule(node)) {
+      return node.getChildAtIndex(1).getString();
+    }
+    for (Node child : node.children()) {
+      String moduleId = getGoogModuleId(child);
+      if (moduleId != null) {
+        return moduleId;
+      }
+    }
+    return null;
+  }
+
+  private Map<String, String> getImportRenameMap(String moduleId, Node node) {
```

Comment:
whose `moduleId` is `moduleId`, the importer or importee? maybe rename the variables?